Tentacle/Content.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  Content.swift
+//  Tentacle
+//
+//  Created by Romain Pouclet on 2016-11-28.
+//  Copyright © 2016 Matt Diephouse. All rights reserved.
+//
+
+import Foundation
+import Argo
+import Curry
+import Runes
+
+public enum Content {
+    public struct File: CustomStringConvertible {
+        public enum ContentType {
+            case file(size: Int, downloadURL: URL?)
+            case directory
+            case symlink(target: String?, downloadURL: URL?)
+            case submodule(url: String?)
```

Comment:
Can you document why these fields are optional? (i.e. that retrieving a directory doesn't get all the fields for all the files)